streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StatefulProcessorNode.java

```diff
@@ -47,11 +47,9 @@ public StatefulProcessorNode(final ProcessorParameters<K, V, ?, ?> processorPara
      * Create a node representing a stateful processor, where the named stores have already been registered.
      */
     public StatefulProcessorNode(final String nodeName,
-                                 final ProcessorParameters<K, V, ?, ?> processorParameters,
-                                 final String[] storeNames) {
+                                 final ProcessorParameters<K, V, ?, ?> processorParameters) {
         super(nodeName, processorParameters);
-
-        this.storeNames = storeNames;
+        this.storeNames = null;
```

Comment:
I don't think we can change this yet since not all StatefulProcessorNode users have been migrated yet. For example the FKJ. Pretty sure this is what's causing the test failures.

Reply:
Mm yeah almost definitely the problem:
```
Caused by: org.apache.kafka.streams.errors.StreamsException: Processor KTABLE-FK-JOIN-SUBSCRIPTION-PROCESSOR-0000000022 has no access to StateStore KTABLE-FK-JOIN-SUBSCRIPTION-STATE-STORE-0000000021 as the store is not connected to the processor. If you add stores manually via '.addStateStore()' make sure to connect the added store to the processor by providing the processor name to '.addStateStore()' or connect them via '.connectProcessorAndStateStores()'. DSL users need to provide the store name to '.process()', '.transform()', or '.transformValues()' to connect the store to the corresponding operator, or they can provide a StoreBuilder by implementing the stores() method on the Supplier itself. If you do not add stores manually, please file a bug report at https://issues.apache.org/jira/projects/KAFKA.
```

basically setting this to null skips the #connectProcessorAndStateStore call, and not every operator that uses this constructor has been migrated to use the `ProcessorParameters#addProcessorTo` method which includes the connection step when adding the state stores